### DISCUSSION TITLE
Add-PnPListItem -List URL  ERROR: The specified list was not found

### DISCUSSION THREAD

**Discussion by myatix** [pnp/powershell#2651] [Q&A]
Hi Everyone,

I am currently running the following in an Azure Function App...

`write-host -f Cyan "Logging Site information in Site Directory for" $webUrl
    $Web = Get-PnPWeb -Connection $webConnect
    $siteDirListUrl = $templateSiteUrl + "/Lists/" + $siteDirList 
    write-host $siteDirListUrl
    Add-PnPListItem -List $siteDirListUrl -Values @{"Title" = $Web.Title; "URL"= $Web.URL; "Site Description"= $Web.Description } -Connection $tplSiteConnect`

When I do write-host $siteDirListUrl I get a valid URL for the list but for some reason the Azure Function Returns the following error:

022-12-14T15:26:06Z   [Error]   ERROR: The specified list was not found. Notice that the title is case sensitive.

Exception             : 
    Type        : System.Management.Automation.PSInvalidOperationException
    ErrorRecord : 
        Exception             : 
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : The specified list was not found. Notice that the title is case sensitive.
            HResult : -2146233087
        CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : InvalidOperation
    TargetSite  : 
        Name          : ProcessRecord
        DeclaringType : PnP.PowerShell.Commands.Base.PnPConnectedCmdlet
        MemberType    : Method
        Module        : PnP.PowerShell.dll
    StackTrace  : 
   at PnP.PowerShell.Commands.Base.PnPConnectedCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPConnectedCmdlet.cs:line 79
   at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord() in c:\build\src\Commands\Base\PnPSharePointCmdlet.cs:line 121
   at System.Management.Automation.Cmdlet.DoProcessRecord()
   at System.Management.Automation.CommandProcessor.ProcessRecord()
    Message     : The specified list was not found. Notice that the title is case sensitive.
    Source      : PnP.PowerShell
    HResult     : -2146233079
CategoryInfo          : InvalidOperation: (:) [Add-PnPListItem], PSInvalidOperationException
FullyQualifiedErrorId : InvalidOperation,PnP.PowerShell.Commands.Lists.AddListItem
InvocationInfo        : 
    MyCommand        : Add-PnPListItem
    ScriptLineNumber : 43
    OffsetInLine     : 5
    HistoryId        : 1
    ScriptName       : C:\home\site\wwwroot\SPOSiteProvisioning\run.ps1
    Line             : Add-PnPListItem -List $siteDirListUrl -Values @{"Title" = $Web.Title; "URL"= $Web.URL; "Site Description"= $Web.Description } -Connection $tplSiteConnect
                       
    PositionMessage  : At C:\home\site\wwwroot\SPOSiteProvisioning\run.ps1:43 char:5
                       +     Add-PnPListItem -List $siteDirListUrl -Values @{"Title" = $Web.Ti …
                       +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    PSScriptRoot     : C:\home\site\wwwroot\SPOSiteProvisioning
    PSCommandPath    : C:\home\site\wwwroot\SPOSiteProvisioning\run.ps1
    InvocationName   : Add-PnPListItem
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, C:\home\site\wwwroot\SPOSiteProvisioning\run.ps1: line 43

**Comment by AndersRask** [comment]
Did you try to just set List parameter to the list object instead of the
url?
The parameter is a pipebind, so it takes different types of objects.

Writing out url might give you a hint as well

ons. 14. dec. 2022 16.32 skrev myatix ***@***.***>:

**Reply by myatix** [reply]
@AndersRask Thanks for the reply! :)

I already did a write host for the URL and it returns to correct list URL. 

> When I do write-host $siteDirListUrl I get a valid URL for the list but for some reason the Azure Function Returns the following error:

According to the PnP documentation I should be able to use Url as a parameter???

https://pnp.github.io/powershell/cmdlets/Add-PnPListItem.html#-list

**Reply by AndersRask** [reply]
Depending on # of items you add, you will get best performance by giving it the list object, so it wont have to fetch it inside the list pipebind.

Also in SharePoint a URL is not just a URL sometimes it has to be relative sometimes not, so URL is not the safest bet.
Titles need to be case sensitive, so both for performance and the above reasons I usually go with list object.

In this case, the correct syntax is this

`Add-PnPListItem -List "/Lists/test" -Values @{"Title"="someValue"}`
